app/models/singly_linked_list.rb

```diff
@@ -0,0 +1,77 @@
+class SinglyLinkedList
+  include Enumerable
```

Comment:
That's way cool!  Well, I suppose...

I'm rusty on infinite closures....I would have thought that a break or a
return were necessary somewhere in there.  Does the yield automatically
break out?  That seems wild, to me.


On Mon, Feb 1, 2016 at 5:53 PM, zfletch <notifications@github.com> wrote:

> In app/models/singly_linked_list.rb
> <https://github.com/dgoldhirsch/little-stuff/pull/3#discussion_r51495039>:
>
> > @@ -0,0 +1,77 @@
> > +class SinglyLinkedList
> > +  include Enumerable
>
> I agree that we shouldn't test #grep, #include?, etc. But the Enumerable
> module is basically designed to work correctly as long as there is an
> #each method defined. From http://ruby-doc.org/core-2.3.0/Enumerable.html:
>
> The Enumerable mixin provides collection classes with several traversal
> and searching methods, and with the ability to sort. The class must provide
> a method each, which yields successive members of the collection.
>
> I think we can trust Enumerable as long as each yields each member of the
> list in turn, it's pretty robust. In fact, it even works with an infinite
> each method. For example:
>
> class Fib
>   include Enumerable
>
>   def each
>     first, second = 0, 1
>     loop do
>       yield second
>       first, second = second, first + second
>     end
>   endend
> Fib.new.first(10) # => [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dgoldhirsch/little-stuff/pull/3/files#r51495039>.
>